recipes/darknet/all/conanfile.py

```diff
@@ -0,0 +1,96 @@
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_opencv": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_opencv": False,
+    }
+    exports_sources = ['patches/*']
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _commit(self):
+        url = self.conan_data["sources"][self.version]["url"]
+        return url[url.rfind("/")+1:url.rfind(".")]
+
+    @property
+    def _lib_to_compile(self):
+        if not self.options.shared:
+            return "$(ALIB)"
+        else:
+            return "$(SLIB)"
+
+    @property
+    def _shared_lib_extension(self):
+        if self.settings.os == "Macos":
+            return ".dylib"
+        else:
+            return ".so"
+
+    def _patch_sources(self):
+        tools.replace_in_file(os.path.join(self._source_subfolder, "Makefile"), "OPENCV=0", "OPENCV={}".format("1" if self.options.with_opencv else "0"))
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "Makefile"),
+            "SLIB=libdarknet.so",
+            "SLIB=libdarknet" + self._shared_lib_extension
+        )
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "Makefile"),
+            "all: obj backup results $(SLIB) $(ALIB) $(EXEC)",
+            "all: obj backup results " + self._lib_to_compile
+        )
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_opencv:
+            # FIXME: opencv recipe is missing on CCI
+            raise ConanInvalidConfiguration("opencv recipe is not (yet) available on CCI")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = self.name + "-" + self._commit
+        os.rename(extracted_folder, self._source_subfolder)
+        for patch in self.conan_data["patches"].get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        self._patch_sources()
+        with tools.chdir(self._source_subfolder):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.make()
+
+    def package(self):
+        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
+        self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+        self.copy("*.so", dst="lib", keep_path=False)
+        self.copy("*.dylib", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
+
+    def package_info(self):
+        self.cpp_info.libs = ["darknet"]
+        if self.settings.os == "Linux":
+            self.cpp_info.system_libs = ["m", "pthread"]
```

Comment:
Because darknet is a c++ library with a c interface.
See https://docs.conan.io/en/latest/reference/tools.html#tools-stdcpp-library

```suggestion
            self.cpp_info.system_libs = ["m", "pthread"]
        if tools.stdcpp_library(self):
            self.cpp_info.system_libs.append(tools.stdcpp_library(self))
```

This change allows converting your test_package.cpp to test_package.c.
I think. 😄

Reply:
@madebr do you know a smart way to detect it? Usually I need to know to project.

Reply:
I don't know of a automated way apart from checking the symbols in the library using `ldd`/`objdump`/`nm`.
You can, by default, create a `test_package.c`.
But it depends on what symbols are used by the test
If the test does only touch c functions, then the linker won't complain.